backend/src/main/java/gov/cdc/usds/simplereport/service/SpecimenService.java

```diff
@@ -0,0 +1,312 @@
+    return loincSnomedResponses;
+  }
+
+  // TODO: Test processInitialSnomedRelations correctly identifies specimens
+  // TODO: Test handling of different relation types in processInitialSnomedRelations
+  private void processAndSaveInitialSnomedRelations(
+      Map<String, List<Specimen>> specimensByLoincSystemCode,
+      Map<String, CompletableFuture<HttpResponse<String>>> loincSnomedResponses) {
+    Map<String, Specimen> deduplicatedLoincSnomedToSpecimen = new HashMap<>();
+    Map<String, SpecimenBodySite> deduplicatedSpecimenCodeSiteCodeToSpecimenBodySite =
+        new HashMap<>();
+
+    // In specimensByLoincSystemCode, each loincSystemCode has a list of specimens due to the
+    // many-to-many relationship
+    // of loincSystemCode to snomedCode for specimens
+    for (String loincSystemCode : specimensByLoincSystemCode.keySet()) {
+      for (Specimen specimen : specimensByLoincSystemCode.get(loincSystemCode)) {
+        CompletableFuture<HttpResponse<String>> responseFuture =
+            loincSnomedResponses.get(
+                "initialRelationRequest-loinc:"
+                    + loincSystemCode
+                    + "-snomed:"
+                    + specimen.getSnomedCode());
+        HttpResponse<String> response = responseFuture.getNow(null);
+        JSONObject body = new JSONObject(response.body());
+        JSONArray results = (JSONArray) body.get("result");
+
+        // Looping through all relationships (of all types) for the snomed specimen
+        for (int i = 0; i < results.length(); i++) {
+          // sometimes there are duplicates within results itself, see this endpoint using snomed
+          // 65216001
+          // https://uts-ws.nlm.nih.gov/rest/content/current/source/SNOMEDCT_US/65216001/relations?apiKey=%s&pageSize=500&additionalRelationLabel=specimen_source_topography_of
+          // and compare the objects where "ui": "R126619655" and "ui": "R123156468"
+          // due to this, we have to deduplicate our loincSystemCode-snomedCode combinations
+          JSONObject result = (JSONObject) results.get(i);
+          boolean saveSnomed = false;
+          String relationLabel = result.get("additionalRelationLabel").toString();
+          // Ensure the snomed specimen is an actual specimen
+          // TODO: Could possibly use an ancestor hierarchy query for the Specimen Concept
+          //    to ensure the snomed specimen concept is a descedant
+          if (specimen.getSnomedDisplay().contains("Specimen")
+              || specimen.getSnomedDisplay().contains("specimen")) {
+            // Here we are including also all the children specimens of the
+            //  snomed specimen concept in question
+            if (StringUtils.equals(relationLabel, "inverse_isa")) {
+              saveSnomed = true;
+              deduplicatedLoincSnomedToSpecimen.put(
+                  specimen.getLoincSystemCode() + specimen.getSnomedCode(), specimen);
```

Comment:
I maybe in too deep here but I feel like this is not saving the child specimens of the "root snomed" but instead saving the same specimen to the map every time we loop through a result
though I don't necessarily think this is blocking merge (what should block merge besides something that breaks the app or causes some changes that are really difficult to reverse?)